autopush/websocket.py

```diff
@@ -512,7 +523,26 @@ def process_hello(self, data):
 
         _, uaid = validate_uaid(uaid)
         self.ps.uaid = uaid
-
+        # Check for the special wakeup commands
+        wake_data = None
+        # If this connection uses the wakeup mechanism, add it.
+        try:
+            wakeup_host = data.get("wakeup_host")
+            mobilenetwork = data.get("mobilenetwork")
+            # Normalize the wake info to a single object.
+            wake_data = dict(data=dict(ip=wakeup_host.get("ip"),
+                             port=wakeup_host.get("port"),
+                             mcc=mobilenetwork.get("mcc", ''),
+                             mnc=mobilenetwork.get("mnc", ''),
+                             netid=mobilenetwork.get("netid", '')))
+            # set this so we can store it for the endpoint to call.
+            self.ps.waker = UDPWake(protocol=self,
+                                    timeout=self.ap_settings.wake_timeout,
+                                    kill_func=self.sendClose, code=4774,
+                                    reason="UDP Idle")
+        except (AttributeError, KeyError):
```

Comment:
Same problem here with the common path. Since UDP will usually not be present, this will add CPU overhead to every single connection establishment. Better to do if/else checks before starting that branch.